crates/holochain/src/core/ribosome/host_fn/block_agent.rs

```diff
@@ -177,4 +179,56 @@ mod test {
         let bob_get2: Option<Record> = bob_conductor.call(&bob, "get_post", action1).await;
         assert!(bob_get2.is_some());
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn hdk_cell_block_adds_block_to_blockspan_database() {
```

Comment:
This test is one of the requirements of the issue #5124.

It tests that a block is added to the database when the HDK call `block_agent` is called. It doesn't test that blocks are respected. I plan to test that for all blocks in the integration tests in the holochain, not specifically for the HDK call.